### PR TITLE
Re: Trust and Go support

### PR DESCRIPTION
**Pull request by doawoo**
Again, picking up the great work from @ConnorRigby --
* locking in the new dep versions
* revising the name of a device's MAC address vs Serial Number
* Add `device_type` argument to settings functions so we use the right slots for each chip type

### CODE REVIEW

**Comment by doawoo** [comment]
Drafted this because I discovered an issue in the settings read/write logic of nerves key with trust and go. Currently we assume slots `[8, 7, 6, 5]` are writeable, but only slot 8 is writeable for general data in Trust and Go chips

File: lib/nerves_key.ex [hunk 1]
```diff
@@ -217,12 +368,19 @@ defmodule NervesKey do
...
+  def provision_aux_certificates(transport, signer_cert, signer_key, type \\ :nerves_key) do
     check_time()
 
-    manufacturer_sn = manufacturer_sn(transport)
+    manufacturer_sn =
+      if type == :nerves_key do
+        manufacturer_sn(transport)
+      else
+        manufacturer_mac(transport, :trust_and_go)
+      end
```
[the remark this answers is not in the document]
Comment (ConnorRigby) [reply lib/nerves_key.ex:380]: i think this is valid, but i should note the `manufacturer_sn` implementation for :trust_and_go isn't actually correct. It pulls
a MAC Address from the cert, which isn't technically correct.

[the remark this answers is not in the document]
Comment (jjcarstens) [reply lib/nerves_key.ex:380]: O! I totally missed these were totally different functions 🤦 ignore me

**Review by ConnorRigby — APPROVED** [review]

**Comment by ConnorRigby** [comment]
:heart: Thanks again!

**Review by jjcarstens — APPROVED** [review]

**Pull request merged**